pkg/analysis/passes/binarypermissions/binarypermissions.go

```diff
@@ -40,67 +46,110 @@ func run(pass *analysis.Pass) (interface{}, error) {
-		}
-		return nil, nil
-	}
-
-	executable := data.Executable
-	executableParentDir := filepath.Join(archiveDir, filepath.Dir(executable))
-	executableName := filepath.Base(executable)
-
-	var foundBinaries = []string{}
-	for _, suffix := range binarySuffixes {
-		binaryPath := filepath.Join(executableParentDir, executableName+suffix)
-		if _, err := os.Stat(binaryPath); err != nil {
+	for pluginJsonPath, data := range metadatamap {
+		if data.Executable == "" {
+			if binaryExecutableFound.ReportAll {
+				binaryExecutableFound.Severity = analysis.OK
+				pass.ReportResult(
+					pass.AnalyzerName,
+					binaryExecutableFound,
+					"No executable defined in plugin.json",
+					fmt.Sprintf("no executable defined in plugin.json: %s", pluginJsonPath),
+				)
+			}
 			continue
 		}
-		foundBinaries = append(foundBinaries, binaryPath)
-	}
-
-	if len(foundBinaries) == 0 {
-		pass.ReportResult(pass.AnalyzerName,
-			binaryExecutableFound,
-			fmt.Sprintf("No binary found for `executable` %s defined in plugin.json", executable),
-			fmt.Sprintf("You defined an executable %s but it could not be found for any of the supported architectures", executable))
-		return nil, nil
-	}
-
-	for _, binaryPath := range foundBinaries {
-
-		// skip windows executables
-		if filepath.Ext(binaryPath) == ".exe" {
-			continue
-		}
-
-		fileInfo, err := os.Stat(binaryPath)
+		relativeTo := filepath.Join(archiveDir, filepath.Dir(pluginJsonPath))
+		executable := data.Executable
+		executableParentDir := filepath.Join(relativeTo, filepath.Dir(executable))
+		executableName := filepath.Base(executable)
+
+		// walk all files in executableParentDir
+		var foundBinaries = []string{}
+		err := filepath.WalkDir(
+			executableParentDir,
+			func(path string, d fs.DirEntry, err error) error {
+				if err != nil {
+					return err
+				}
+				if d.IsDir() {
+					return nil
+				}
+				if d.Type().IsRegular() {
+					if strings.HasPrefix(d.Name(), executableName) {
+						foundBinaries = append(foundBinaries, path)
+					}
+				}
+				return nil
+			},
+		)
 		if err != nil {
-			pass.ReportResult(pass.AnalyzerName,
-				binaryExecutablePermissions,
-				fmt.Sprintf("Could not read permissions for executable %s", binaryPath),
-				fmt.Sprintf("Could not read the file %s. This could be too few permissions in your binary files or your zip file is corrupted", binaryPath))
-			continue
+			pass.ReportResult(
+				pass.AnalyzerName,
+				archiveFilesError,
+				"error reading your archive files",
+				fmt.Sprintf("error reading your archive files: %s", err.Error()),
+			)
+		}
+		if len(foundBinaries) == 0 {
+			pass.ReportResult(
+				pass.AnalyzerName,
+				binaryExecutableFound,
+				fmt.Sprintf(
+					"No binary found for `executable` %s defined in %s", executable, pluginJsonPath,
+				),
+				fmt.Sprintf(
+					"You defined an executable %s but it could not be found for any of the supported architectures",
+					executable,
+				),
+			)
+			return nil, nil
```

Comment:
Also here I think this should be a `continue` instead?

Reply:
Yes! thanks for catching these. Added.